bazel/pbgen/main.go

```diff
@@ -1959,12 +1946,46 @@ func nameToCppNamespace(file protoreflect.FileDescriptor) string {
 	return strings.ReplaceAll(string(file.Package()), ".", "::")
 }
 
+func mapImport(path string) string {
+	if strings.HasPrefix(path, "proto/redpanda/pbgen") {
+		return ""
+	}
+	switch path {
+	case "google/protobuf/duration.proto",
+		"google/protobuf/timestamp.proto":
+		return "absl/time/time.h"
+	case "google/protobuf/field_mask.proto":
+		return "serde/protobuf/field_mask.h"
+	default:
+		if strings.HasPrefix(path, "google/protobuf") {
+			log.Fatalf("well known import %q must be handled specially in code generation\n", path)
+		}
+		return path + ".h"
+	}
+}
+
+func isWellKnownType(d protoreflect.Descriptor) bool {
+	switch d.FullName() {
+	case "google.protobuf.Duration",
+		"google.protobuf.FieldMask",
+		"google.protobuf.Timestamp":
+		return true
+	}
+	return false
+}
+
 func fullyQualifiedTypeName(d protoreflect.Descriptor) string {
 	switch d.FullName() {
 	case "google.protobuf.Duration":
 		return "absl::Duration"
+	case "google.protobuf.Timestamp":
+		return "absl::Time"
 	case "google.protobuf.FieldMask":
 		return "serde::pb::field_mask"
+	default:
+		if isWellKnownType(d) {
+			log.Fatalf("well-known types need an entry above: %s\n", d.FullName())
+		}
```

Comment:
Now in the future this is all that needs to be modified to support well known types.